pkg/deploy/controller/deployment/controller.go

```diff
@@ -24,40 +39,48 @@ import (
 //
 //   1. If the deployment finished normally, the deployer pod is deleted.
 //   2. If the deployment failed, the deployer pod is not deleted.
-//
-// Use the DeploymentControllerFactory to create this controller.
 type DeploymentController struct {
-	// serviceAccount to create deployment pods with
+	// rn is used for updating replication controllers.
+	rn kclient.ReplicationControllersNamespacer
+	// pn is used for creating, updating, and deleting deployer pods.
+	pn kclient.PodsNamespacer
+
+	// queue contains replication controllers that need to be synced.
+	queue workqueue.RateLimitingInterface
+
+	// rcStore is a store of replication controllers.
+	rcStore cache.StoreToReplicationControllerLister
+	// podStore is a store of pods.
+	podStore cache.StoreToPodLister
+	// rcStoreSynced makes sure the rc store is synced before reconcling any deployment.
+	rcStoreSynced func() bool
+	// podStoreSynced makes sure the pod store is synced before reconcling any deployment.
+	podStoreSynced func() bool
+
+	// deployerImage specifies which Docker image can support the default strategies.
+	deployerImage string
+	// serviceAccount to create deployment pods with.
 	serviceAccount string
-	// deploymentClient provides access to deployments.
-	deploymentClient deploymentClient
-	// podClient provides access to pods.
-	podClient podClient
-	// makeContainer knows how to make a container appropriate to execute a deployment strategy.
-	makeContainer func(strategy *deployapi.DeploymentStrategy) *kapi.Container
-	// decodeConfig knows how to decode the deploymentConfig from a deployment's annotations.
-	decodeConfig func(deployment *kapi.ReplicationController) (*deployapi.DeploymentConfig, error)
+	// environment is a set of environment variables which should be injected into all
+	// deployer pod containers.
+	environment []kapi.EnvVar
+	// codec is used for deserializing deploymentconfigs from replication controller
+	// annotations.
+	codec runtime.Codec
 	// recorder is used to record events.
 	recorder record.EventRecorder
 }
 
-// fatalError is an error which can't be retried.
-type fatalError string
-
-func (e fatalError) Error() string { return "fatal error handling deployment: " + string(e) }
-
-// actionableError is an error on which users can act
-type actionableError string
-
-func (e actionableError) Error() string { return string(e) }
-
 // Handle processes deployment and either creates a deployer pod or responds
 // to a terminal deployment status.
 func (c *DeploymentController) Handle(deployment *kapi.ReplicationController) error {
 	currentStatus := deployutil.DeploymentStatusFor(deployment)
 	nextStatus := currentStatus
 	deploymentScaled := false
 
+	deployerPodName := deployutil.DeployerPodNameForDeployment(deployment.Name)
+	deployer, deployerErr := c.pn.Pods(deployment.Namespace).Get(deployerPodName)
```

Comment:
> This is the deployer pod and I am not sure I want to make it susceptible to a stale cache. Especially when we collapse the deployer pod controller on this controller, we will need its latest status in order to make calls for the status of the deployment. Isn't this a valid concern for having it live?

Live checks with multiple etcds are surprisingly unreliable.  Didn't we get here from either a pod update (you really want to use a matching cache, otherwise you get an update, do a get, hit a stale etcd, and get a 404) or an RC (we'll get a pod update later)?